File: spec/test_positions.c
#include "testparser.h"
#include <assert.h>
#include <string.h>
#include <stdio.h>

int main()
{
    char const * input = "    Hello\n\n        4200\n";
    p_context_t * context;
    context = p_context_new((uint8_t const *)input, strlen(input));
    assert(p_parse(context) == P_SUCCESS);
    p_context_delete(context);

    printf("\n");

    input = "\n tok2";
    context = p_context_new((uint8_t const *)input, strlen(input));
    assert(p_parse(context) == P_SUCCESS);
    p_context_delete(context);

    printf("\n");

    input = "  tok1";
    context = p_context_new((uint8_t const *)input, strlen(input));
    assert(p_parse(context) == P_SUCCESS);
    p_context_delete(context);

    return 0;
}

File: spec/test_positions.rs
use testparser::*;

fn main() {
    let mut c = p_context_new(b"    Hello\n\n        4200\n");
    assert_eq!(P_SUCCESS, p_parse(&mut c));
    p_context_delete(c);

    println!();

    let mut c = p_context_new(b"\n tok2");
    assert_eq!(P_SUCCESS, p_parse(&mut c));
    p_context_delete(c);

    println!();

    let mut c = p_context_new(b"  tok1");
    assert_eq!(P_SUCCESS, p_parse(&mut c));
    p_context_delete(c);
}
